src/function.py

```diff
@@ -32,7 +32,27 @@
 class ArgCacheDecorator(ArgCache):
     """ An ArgCache that gets its parameters from a function. """
 
-    def __init__(self, func, *args, **kwargs):
```

Comment:
I'm not sure. As far as I can tell there were no sensible other args you'd want to pass here.